src/Aspire.Hosting.Docker/api/Aspire.Hosting.Docker.cs

```diff
@@ -0,0 +1,545 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by a tool.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+namespace Aspire.Hosting
+{
+    public static partial class DockerComposePublisherExtensions
+    {
+        public static IDistributedApplicationBuilder AddDockerComposePublisher(this IDistributedApplicationBuilder builder, System.Action<Docker.DockerComposePublisherOptions>? configureOptions = null) { throw null; }
+
+        public static IDistributedApplicationBuilder AddDockerComposePublisher(this IDistributedApplicationBuilder builder, string name, System.Action<Docker.DockerComposePublisherOptions>? configureOptions = null) { throw null; }
+    }
+}
+
+namespace Aspire.Hosting.Docker
+{
+    public sealed partial class DockerComposePublisherOptions : Publishing.PublishingOptions
+    {
+        public string? DefaultContainerRegistry { get { throw null; } set { } }
+
+        public string? ExistingNetworkName { get { throw null; } set { } }
+    }
+}
+
+namespace Aspire.Hosting.Docker.Resources
+{
+    [YamlDotNet.Serialization.YamlSerializable]
+    public sealed partial class ComposeFile
```

Comment:
Why is this class public?

Reply:
The plan is to expose this all to allow people to tweak but we won't mark this a pre-release until we figure out the pattern.